src/runtime/mprof.go

```diff
@@ -1422,6 +1484,8 @@ func goroutineProfileWithLabelsConcurrent(p []profilerecord.StackRecord, labels
 	goroutineProfile.active = true
 	goroutineProfile.records = p
 	goroutineProfile.labels = labels
+	goroutineProfile.infos = infos
+
```

Comment:
don't you need to also update `doRecordGoroutineProfile` so that it updates `goroutineProfile.infos`?